mongo/client.go

```diff
@@ -776,7 +776,7 @@ func (c *Client) ListDatabaseNames(ctx context.Context, filter interface{}, opts
 //
 // Any error returned by the fn callback will be returned without any modifications.
 func WithSession(ctx context.Context, sess Session, fn func(SessionContext) error) error {
-	return fn(NewSessionContext(ctx, sess))
+	return fn(ContextWithSession(ctx, sess))
```

Comment:
I can leave the function named as NewSessionContext and rename the references in the baas repo instead (there are not that many). Not sure how we have previously done this but removing these renamings will keep us more 1 to 1 with the main driver